code-journal/src/FormElement.tsx

```diff
@@ -0,0 +1,183 @@
+import { FormEvent, useEffect, useRef, useState } from 'react';
+import {
+  addEntry,
+  Entry,
+  readEntry,
+  removeEntry,
+  updateEntry,
+} from './data.ts';
+import { useNavigate, useParams } from 'react-router-dom';
+import { createPortal } from 'react-dom';
+const imgSrc = '/placeholder-image-square.jpg';
+
+export function FormElement() {
+  const [newTitle, setNewTitle] = useState<string>();
+  const [newImg, setNewImg] = useState<string>();
+  const [newNote, setNewNote] = useState<string>();
+  const [entryEdit, setEntryEdit] = useState<Entry>();
+  const [deleteConfirmDisplayed, setDeleteConfirmDisplayed] = useState(false);
+  const { id } = useParams();
+  const imgRef = useRef<HTMLImageElement>(null);
+  const navigate = useNavigate();
+  function handleEntrySubmission(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    if (!entryEdit) {
+      addEntry({
+        title: newTitle ?? '',
+        notes: newNote ?? '',
+        photoUrl: newImg ?? '',
+      });
+    } else {
+      updateEntry({
+        title: newTitle ?? '',
+        notes: newNote ?? '',
+        photoUrl: newImg ?? '',
+        entryId: entryEdit.entryId,
+      });
+    }
+    navigate('/entries');
+  }
+  useEffect(() => {
+    const changeImage = async function (): Promise<void> {
+      try {
+        const fgh = await new Promise((resolve, reject) => {
```

Comment:
Why is this a promise? I'm guessing an AI generated junky code for you again.

All you need is the onload and onerror functions at line 46 and 47. And those don't do anything meaningful since they don't update state. The implementation of those functions should update state instead of calling resolve/reject.

I don't mind you using AI to solve difficult problems, but as you can see, if you don't understand the code it generates you will often end up with either ugly code or useless code!